File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseConversationModel.cs
using KnowledgeSharingApi.Domains.Models.Entities;
using KnowledgeSharingApi.Domains.Models.Entities.Tables;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseConversationModel : Conversation
    {
        public List<ResponseParticipantModel> Participants { get; set; } = [];

        public List<ResponseMessageModel> Messages { get; set; } = [];


        protected override ResponseConversationModel Init()
        {
            return new ResponseConversationModel();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-conversation-model.js


import Conversation from '../entities/conversation';
import ResponseMessageModel from './response-message-model';
import ResponseParticipantModel from './response-participant-model';

class ResponseConversationModel extends Conversation {
    constructor() {
        super();
        this.Participants = null;
        this.Messages = null;
    }

    init() {
        return new ResponseConversationModel();
    }

    copy(entity){
        super.copy(entity);
        if (this.Participants === null) {
            this.Participants = [];
        } else {
            this.Participants = this.Participants.map(item => {
                return new ResponseParticipantModel().copy(item);
            });
        }
        if (this.Messages === null) {
            this.Messages = [];
        } else {
            this.Messages = this.Messages.map(item => {
                return new ResponseMessageModel().copy(item);
            });
        }
        return this;
    }
}

export default ResponseConversationModel;
